department-of-veterans-affairs/caseflow
# Add/edit unscheduled hearing notes in Case Details

Resolves https://vajira.max.gov/browse/CASEFLOW-500

Part 1 of stack - proceeds https://github.com/department-of-veterans-affairs/caseflow/pull/16164
Note: Jest failures are fixed in following PR.

### Description
- Create a new component `EditUnscheduledNotesModal` 
- Update `UnscheduledNotes` component to display readonly notes in Case Details
- Update `CaseDetailsView` to pass hearingTasks as prop to `CaseHearingsDetails`
- Update logic in `CaseHearingsDetails` to display unscheduled hearings as well as update the order for scheduled hearings to be in desc order

### Acceptance Criteria
- [x] For any appeal with an open HearingTask, on the Case Details page the Hearings section appears even if there is no scheduled hearing yet
- [x]  If the appeal has an open schedule hearing task, display at the top of the Hearings section 
  - [x] Unscheduled Hearing header
  - [x] "Type:" and hearing type 
  - [x] "Notes:" label and Edit button
- [x] If the hearing does not have an open schedule hearing task, only displayed any scheduled hearings
- [x] If the appeal has an open schedule hearing task and previously scheduled hearings, show both the Unscheduled Hearing header and notes as well as the previously scheduled hearing info
- [x] Previously scheduled hearings should be ordered in descending order of hearing dates (i.e., most recent first)
- [x] When a user clicks the Edit button, display a modal where a user can enter notes (CASEFLOW-499) and save their notes
- [x] When a user clicks Save
   - [x] They're returned to the Case Details page
   - [x] A success or error alert is displayed at the top of the page
   - [x] The added/edited notes are shown under the Notes label and Edit button
   - [x] The following Google Analytics event is sent
- [x] If a user has previously entered notes
  - [x] They're shown under the Notes label and Edit button when a user first loads the Case Details page
  - [x] They populate the modal that opens when a user clicks the edit button 
 
### Testing Plan
1. Login as `BVASYELLOW`
2. Go to http://localhost:3000/hearings/schedule/assign or https://hearings.caseflowdemo.com/hearings/schedule/assign
3. Pick an appeal from any RO and go to Case details page for it
4. Scroll down to the hearing section and ensure that there exists an unscheduled hearing section. Interact with it.
5. If the appeal does not have a previously scheduled hearing, you can create one by scheduling the veteran for hearing and refreshing the page.  Notice that the unscheduled hearing section no longer displays. 
6. Postpone the hearing by clicking on `Postpone Hearing` in the actions menu but instead of rescheduling, click `Send to Schedule Veteran list` and hit enter. Scroll down to the hearing section and ensure that you can see both Unscheduled and and Scheduled hearings section.


### User Facing Changes
<img width="580" alt="Screen Shot 2021-04-23 at 11 12 14 AM" src="https://user-images.githubusercontent.com/20735998/115892031-c4a72b80-a424-11eb-97f7-16083dad332d.png">|<img width="569" alt="Screen Shot 2021-04-23 at 10 50 13 AM" src="https://user-images.githubusercontent.com/20735998/115892081-d25cb100-a424-11eb-837f-a512e23e12dd.png">
--|--
